table.cc

```diff
@@ -187,12 +189,12 @@ table::make_reader(schema_ptr s,
         // - fast forwarding is implemented in reversed sstable readers.
         readers.emplace_back(_cache.make_reader(s, permit, range, slice, pc, std::move(trace_state), fwd, fwd_mr));
     } else {
-        readers.emplace_back(make_sstable_reader(s, permit, _sstables, range, slice, pc, std::move(trace_state), fwd, fwd_mr));
+        readers.emplace_back(maybe_reverse(
+                make_sstable_reader(maybe_reverse(s, slice), permit, _sstables, range, slice, pc, std::move(trace_state), fwd, fwd_mr),
+                slice, max_memory_for_unlimited_query));
```

Comment:
And similarly, we could have made the sstable reader and memtable reader call make_reversing_reader() so they can undo  it locally when they gain the ability.

Reply:
That's the idea. I do it in following commits for sstable reader. For memtable reader I imagine @enedil's patchset could do this when rebased on top of mine (I'll do the rebase).

Reply:
Yes, I saw it later and forgot to acknowledge.